test/rulesets/Base/mapreduce.jl

```diff
@@ -275,57 +286,67 @@ end
     end  # cumprod
 
     @testset "accumulate(f, ::Array)" begin
+        # `accumulate(f, A; init)` goes to `_accumulate!(op, B, A, dims::Nothing, init::Nothing)`. 
+        # The rule is now attached there, as this is the simplest way to handle `init` keyword.
+        @eval using Base: _accumulate!
+
         # Simple
-        y1, b1 = rrule(CFG, accumulate, *, [1, 2, 3, 4]; init=1)
+        y1, b1 = rrule(CFG, _accumulate!, *, [0, 0, 0, 0], [1, 2, 3, 4], nothing, Some(1))
         @test y1 == [1, 2, 6, 24]
-        @test b1([1, 1, 1, 1]) == (NoTangent(), NoTangent(), [33, 16, 10, 6])
+        @test b1([1, 1, 1, 1])[3] isa ChainRulesCore.NotImplemented
+        @test b1([1, 1, 1, 1])[4] == [33, 16, 10, 6]
+        @test b1([1, 1, 1, 1])[6] isa Tangent{Some{Int64}}
```

Comment:
Maybe fixes CI?
```suggestion
        @test b1([1, 1, 1, 1])[6] isa Tangent{<:Some{Int64}}
```
Also, why on earth does `accumulate` use nothing/Some for `init`, while `foldl` uses `_InitialValue`?

Reply:
No, it needs to be:
```suggestion
        @test b1([1, 1, 1, 1])[6] isa Tangent{Some{Int}}
```